js/main.js

```diff
@@ -14,11 +15,29 @@ if(pokedexStorageJSON!==null){
     entry.appendChild(pokeballIcon)
     entry.appendChild(pokemonName)
     list.appendChild(entry)
-  })}
+    entry.addEventListener('click',()=>{highlightEntry(entry)})
+    searchBar.addEventListener('input',()=>{
+      handleSearch(entry)
+        }
+      )
+    }
+  )
+}
 
 const listItems=document.querySelectorAll('li')
-list.addEventListener('click',(event)=>{
-  event.target.closest('li').className='selected'
-  listItems.forEach((element)=>{if (event.target.closest('li')!==element){
-    element.className=''
-  }})})
+  function highlightEntry(clicked){
+    listItems.forEach((element)=>{
+      element.classList.remove('selected')
+      }
+    )
+    clicked.classList.add('selected')
+  }
+
+function handleSearch(entry){
+  if(entry.textContent.toLowerCase().includes(searchBar.value.toLowerCase())){
+        entry.classList.remove('hidden')
+        }
+      else{
+        entry.classList.add('hidden')
+      }
+    }
```

Comment:
This is a clever solution! However, you want to use an API endpoint!

Reply:
I looked at the docs, and I found this potential endpoint: `https://pokeapi.co/api/v2/pokemon/{id or name}/`